js/main.js

```diff
@@ -0,0 +1,40 @@
+var $newEntry = document.querySelector('form');
+var $previewPhoto = document.querySelector('img');
+var $photoURL = document.querySelector('#photo');
+$photoURL.addEventListener('input', photoURLBox);
+
+function photoURLBox(event) {
+  $previewPhoto.setAttribute('src', $newEntry.elements.photo.value);
+}
+
+function newEntry(event) {
+  event.preventDefault();
+  var title = $newEntry.elements.title.value;
+  var photoURL = $newEntry.elements.photo.value;
+  var notes = $newEntry.elements.notes.value;
+  var newEntryObject = {
+    title,
+    photoURL,
+    notes,
+    entryId: data.nextEntryId
+  };
+  data.nextEntryId++;
+  data.entries.unshift(newEntryObject);
+  $previewPhoto.setAttribute('src', 'images/placeholder-image-square.jpg');
+  $newEntry.reset();
+}
+
+$newEntry.addEventListener('submit', newEntry);
+
+window.addEventListener('beforeunload', stringified);
+
+function stringified(event) {
+  var newEntryJSON = JSON.stringify(data.entries);
+  localStorage.setItem('new-entry-storage', newEntryJSON);
+}
+
+var previousEntryJSON = localStorage.getItem('new-entry-storage');
+
+if (previousEntryJSON !== null) {
+  data.entries = JSON.parse(previousEntryJSON);
+}
```

Comment:
Any code dealing with localStorage should be in `data.js`, since that is where the variable is that you are preserving.

Also, you want to save the entire `data` variable in localStorage, not just `data.entries`. Make sure you reset localStorage after this change so the format of your data doesn't get botched.